core/events/events_procedure_enable.ts

```diff
@@ -5,13 +5,42 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import type {IProcedureModel} from '../interfaces/i_procedure_model.js';
 import * as registry from '../registry.js';
+import type {Workspace} from '../workspace.js';
 
 import {ProcedureBase} from './events_procedure_base.js';
 import * as eventUtils from './utils.js';
 
 
-export class ProcedureEnable extends ProcedureBase {}
+export class ProcedureEnable extends ProcedureBase {
+  type = eventUtils.PROCEDURE_ENABLE;
+
+  private oldState: boolean;
+  private newState: boolean;
+
+  constructor(workspace: Workspace, model: IProcedureModel) {
+    super(workspace, model);
+
+    this.oldState = !model.getEnabled();
+    this.newState = model.getEnabled();
```

Comment:
Is it really necessary to store both?  Do you foresee in the future emitting events when the enable status "changes" from `true` to `true`, or from `false` to `false`?

Reply:
I only included both for readability of the `run` function. Otherwise you have to do:

```
    if (forward) {
      procedureModel.setEnabled(this.newState);
    } else {
      procedureModel.setEnabled(!this.newState);
    }
```

Which I think is much more confusing than:

```
    if (forward) {
      procedureModel.setEnabled(this.newState);
    } else {
      procedureModel.setEnabled(this.oldState);
    }
```

Reply:
I admit that having learned to program on a machine with 1kB of memory (and having done most of my university CS assignments on a machine with 100MB of RAM _shared between 60–100 simultaneous users_) might make me excessively concerned about wasting memory, but it does seem kind of distasteful to store the information twice just to make `run` very slightly more readable.